collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/CustomDatePickerDialog.java

```diff
@@ -16,97 +16,55 @@
 
 package org.odk.collect.android.fragments.dialogs;
 
+import android.app.DatePickerDialog;
 import android.app.Dialog;
 import android.content.Context;
-import android.content.DialogInterface;
 import android.os.Bundle;
 import androidx.fragment.app.DialogFragment;
 import androidx.appcompat.app.AlertDialog;
 import android.view.View;
+import android.widget.DatePicker;
 import android.widget.NumberPicker;
 import android.widget.TextView;
 
-import org.javarosa.core.model.FormIndex;
 import org.joda.time.LocalDateTime;
 import org.joda.time.chrono.GregorianChronology;
 import org.odk.collect.android.R;
-import org.odk.collect.android.application.Collect;
 import org.odk.collect.android.logic.DatePickerDetails;
-import org.odk.collect.android.javarosawrapper.FormController;
 import org.odk.collect.android.utilities.DateTimeUtils;
+import org.odk.collect.android.widgets.utilities.DateTimeWidgetUtils;
 
 /**
  * @author Grzegorz Orczykowski (gorczykowski@soldevelo.com)
  */
 public abstract class CustomDatePickerDialog extends DialogFragment {
```

Comment:
You still didn't implement view model as I said before (too keep data) so your picker will reset values for example if your rotate your device. This is something we should test automatically.

Reply:
You implemented the viewmodel but you don't use values stored there so the result is the same once you rotate your device everything is reset. As above would be good to have a test for it.

Reply:
I tested it on my device. I didn't notice values getting reset on the device rotation

Reply:
I think it happens only in custom date pickers like coptic/ethiopian etc. Did you test them?

Reply:
Oh right I need to take care of the dates displayed in the textview of the dialog.

Reply:
Should I write the test for `CusotmDatePickerDialog` as I ddin't use the viewmodel to set the values shown on the dialog on screen orientation change.

Reply:
It would be good to have a test that confirms everything is fine after recreation.

Reply:
That would mean writing tests for all CustomDatePickerDialogs, should I write them as part of this PR itself?

Reply:
I wanted to say please file a separate issue but it would be great to have at least those that check if values don change on recreation so it shouldn't be a big deal.

Reply:
Oh sure! I will add tests for that as part of this PR, and all the other tests as a different PR, after filing a separate issue.